all_bin_scenario.py

```diff
@@ -253,31 +236,41 @@ def max_stepsize(t: float) -> float:
                     folder=f"mb{nb_bin+1}_{sub_bin.mode}_results",
                 )
-            elif sub_bin.material == "B":
-                my_model, quantities = make_B_mb_model(
-                    temperature=T_function_B,
-                    deuterium_ion_flux=deuterium_ion_flux,
-                    tritium_ion_flux=tritium_ion_flux,
-                    deuterium_atom_flux=deuterium_atom_flux,
-                    tritium_atom_flux=tritium_atom_flux,
-                    # FIXME: -1s here to avoid last time step spike
-                    final_time=my_scenario.get_maximum_time() - 1,
-                    L=sub_bin.thickness,
-                    folder=f"mb{nb_bin+1}_{sub_bin.mode}_results",
-                )
+        elif material == "B":
+            my_model, quantities = make_B_mb_model(
+                temperature=T_function_B,
+                deuterium_ion_flux=deuterium_ion_flux,
+                tritium_ion_flux=tritium_ion_flux,
+                deuterium_atom_flux=deuterium_atom_flux,
+                tritium_atom_flux=tritium_atom_flux,
+                # FIXME: -1s here to avoid last time step spike
+                final_time=my_scenario.get_maximum_time() - 1,
+                L=sub_bin.thickness,
+                folder=f"mb{nb_bin+1}_{sub_bin.mode}_results",
+            )
 
-            elif sub_bin.material == "SS":
-                my_model, quantities = make_DFW_mb_model(
-                    temperature=T_function_W,  # TODO: update for DFW function
-                    deuterium_ion_flux=deuterium_ion_flux,
-                    tritium_ion_flux=tritium_ion_flux,
-                    deuterium_atom_flux=deuterium_atom_flux,
-                    tritium_atom_flux=tritium_atom_flux,
-                    # FIXME: -1s here to avoid last time step spike
-                    final_time=my_scenario.get_maximum_time() - 1,
-                    L=sub_bin.thickness,
-                    folder=f"mb{nb_bin+1}_dfw_results",
-                )
+        elif material == "SS":
+            my_model, quantities = make_DFW_mb_model(
+                temperature=T_function_W,  # TODO: update for DFW function
+                deuterium_ion_flux=deuterium_ion_flux,
+                tritium_ion_flux=tritium_ion_flux,
+                deuterium_atom_flux=deuterium_atom_flux,
+                tritium_atom_flux=tritium_atom_flux,
+                # FIXME: -1s here to avoid last time step spike
+                final_time=my_scenario.get_maximum_time() - 1,
+                L=sub_bin.thickness,
+                folder=f"mb{nb_bin+1}_dfw_results",
+            )
+
+        return my_model, quantities
+
+    ############# RUN FW BIN SIMUS #############
+    # TODO: adjust to run monoblocks in parallel
+    # for fw_bin in FW_bins.bins:
+    for fw_bin in range(3): # only running 3 fw_bins to demonstrate capability 
+        nb_bin = FW_bins.get_bin(fw_bin)
+        for sub_bin in nb_bin.sub_bins:
```

Comment:
You're getting mixed up between `nb_bin` (which should be an int by denomination) and `fw_bin` (which should be a bin). This is much more straightforward:
```suggestion
    for fw_bin in FW_bins.bins[:3]: # only running 3 fw_bins to demonstrate capability
        for sub_bin in fw_bin.sub_bins:
```